guides/source/working_with_javascript_in_rails.md

```diff
@@ -3,561 +3,212 @@
-
-NOTE: Because submitting forms with HTTP methods other than GET and POST isn't
-widely supported across browsers, all other HTTP methods are actually sent over
-POST with the intended method indicated in the `_method` parameter. Rails
-automatically detects and compensates for this.
-
-#### `data-url` and `data-params`
-
-Certain elements of your page aren't actually referring to any URL, but you may want
-them to trigger Ajax calls. Specifying the `data-url` attribute along with
-the `data-remote` one will trigger an Ajax call to the given URL. You can also
-specify extra parameters through the `data-params` attribute.
-
-This can be useful to trigger an action on check-boxes for instance:
-
-```html
-<input type="checkbox" data-remote="true"
-    data-url="/update" data-params="id=10" data-method="put">
+```bash
+yarn --version
 ```
 
-#### `data-type`
-
-It is also possible to define the Ajax `dataType` explicitly while performing
-requests for `data-remote` elements, by way of the `data-type` attribute.
+If it says something like 1.22.0, Yarn has been installed correctly.
 
-### Confirmations
+Choosing Between Import Maps and a JavaScript Bundler
+————————————
+When you create a new Rails application, you will need to choose between import maps and a JavaScript bundling solution. Every application has different requirements, and you should consider your requirements carefully before choosing a JavaScript option. The choice that you make is not permanent; however, migrating from one option to another may be time consuming for large, complex applications.
 
-You can ask for an extra confirmation of the user by adding a `data-confirm`
-attribute on links and forms. The user will be presented with a JavaScript `confirm()`
-dialog containing the attribute's text. If the user chooses to cancel, the action
-doesn't take place.
+Import maps are the default option because the Rails team believes in import maps potential for reducing complexity, improving developer experience, and delivering performance gains.
 
-Adding this attribute on links will trigger the dialog on click, and adding it
-on forms will trigger it on submit. For example:
+For many applications, especially those that rely primarily on the Hotwire stack for their JavaScript needs, import maps will be the right option for the long term. You can read more about the reasoning behind making import maps the default in Rails 7 [here](https://world.hey.com/dhh/rails-7-will-have-three-great-answers-to-javascript-in-2021-8d68191b).
 
-```erb
-<%= link_to "Dangerous zone", dangerous_zone_path,
-  data: { confirm: 'Are you sure?' } %>
-```
+Other applications may still need a traditional JavaScript bundler. Requirements that indicate that you should choose a traditional bundler include:
+* If your code reqiures a transpilation step, such as JSX or TypeScript.
+* If you need to use JavaScript libraries that include CSS.
```

Comment:
I'm not reallly sure what this line means - do you have an example?

Reply:
Something like [this](https://react-day-picker.js.org/docs/getting-started) which packages some CSS along side a React component that must be imported along with the JS.

I may be incorrect in my assumption, but I think in those cases import maps won't work, or will require a significant amount of effort to get the package functioning.

Reply:
Perhaps generalize this to encompass all webpack loaders: https://webpack.js.org/loaders/

Reply:
Updated to include a general note about reliance on webpack loaders